unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -7051,80 +7116,98 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
                 ItemInHand.gameObject.SetActive(true);
             }
 
-#if UNITY_EDITOR
-            Debug.Log(goodLocationsDict["x"].Count);
-            Debug.Log(goodLocationsDict["x"]);
-#endif
-
-            actionFinished(true, goodLocationsDict);
-        } 
-        
-        public int NumberOfPositionsFromWhichItemIsVisibleHelper(SimObjPhysics theObject, Vector3[] positions) {
-            bool wasStanding = isStanding();
-            Vector3 oldPosition = transform.position;
-            Quaternion oldRotation = transform.rotation;
-
-            if (ItemInHand != null) {
-                ItemInHand.gameObject.SetActive(false);
-            }
-
-            int numTimesVisible = 0;
-            for (int j = 0; j < 2; j++) { // Standing / Crouching
-                if (j == 0) {
-                    stand();
-                } else {
-                    crouch();
-                }
-                for (int i = 0; i < 4; i++) { // 4 rotations
-                    transform.rotation = Quaternion.Euler(new Vector3(0.0f, 90.0f * i, 0.0f));
-                    foreach (Vector3 p in positions) {
-                        transform.position = p;
+            #if UNITY_EDITOR
+                Debug.Log(validAgentPoses.Count);
+                Debug.Log(validAgentPoses);
+            #endif
 
-                        if (objectIsCurrentlyVisible(theObject, 1000f)) {
-                            numTimesVisible += 1;
-                        }
-                    }
-                }
+            if (markActionFinished) {
+                actionFinished(success: true, actionReturn: validAgentPoses);
             }
 
-            if (wasStanding) {
-                stand();
-            } else {
-                crouch();
-            }
-            transform.position = oldPosition;
-            transform.rotation = oldRotation;
-            if (ItemInHand != null) {
-                ItemInHand.gameObject.SetActive(true);
-            }
-
-#if UNITY_EDITOR
-            Debug.Log(4 * 2 * positions.Length);
-            Debug.Log(numTimesVisible);
-#endif
+            return validAgentPoses;
+        }
 
-            return numTimesVisible;
+        // Get the poses with which the agent can interact with 'objectId'
+        // @rotations: if rotation is not specified, we use rotateStepDegrees, which results in [0, 90, 180, 270] by default.
+        public void GetInteractablePoses(
```

Comment:
Note that I have changed this action from `GetInteractablePositions` to `GetInteractablePoses` because it's not only returning and checking for `x/y/z` positions, rather, it checks for the entire pose of the agent.